KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift

```diff
@@ -118,7 +125,7 @@ final class KarhooNewCheckoutViewModel: ObservableObject {
     // MARK: Get simple data to display
 
     func getDateScheduledDescription() -> String {
-        let date = trip?.dateScheduled ?? Date()
+        let date = journeyDetails.scheduledDate ?? Date()
```

Comment:
is that mean that you will not provide trip object to KarhooNewCheckoutViewModel at all?

Reply:
The trip does not exist until we actually make a booking. The view model should contain the trip for the soul purpose that it needs to be returned in the checkout component callback, but we should avoid using info from it to avoid muddying the logic.

Reply:
Based on the old checkout implementation, the `TripInfo` object is set during confirmation flow. It's not required to prepare for the booking. Maybe it will not be required at all (🤞 )